### PR TITLE
SOC-3836 Fix failing tests for reporting posts

### PR DESCRIPTION
**Pull request by slayful**
https://wikia-inc.atlassian.net/browse/SOC-3836

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java [hunk 1]
```diff
@@ -214,12 +215,9 @@ public boolean isStringInURL(String givenString) {
...
     changeImplicitWait(250, TimeUnit.MILLISECONDS);
     try {
-      new WebDriverWait(driver, timeOut).until(new ExpectedCondition<Boolean>() {
-        @Override
-        public Boolean apply(WebDriver driver) {
-          return driver.getCurrentUrl().toLowerCase().contains(givenString.toLowerCase());
-        }
-      });
+      new WebDriverWait(driver, timeOut).until(
+          (ExpectedCondition<Boolean>) driver -> driver.getCurrentUrl().toLowerCase()
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Rename "driver" which hides the field declared at line 45. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostsListPage.java [hunk 2]
```diff
@@ -95,13 +96,8 @@ public void waitForPageReloadWith(final String categoryName) {
...
       new FluentWait<>(getPost())
           .withTimeout(DiscussionsConstants.TIMEOUT, TimeUnit.SECONDS)
-          .until(new Predicate<Post>() {
-            @Override
-            public boolean apply(@Nullable Post post) {
-              return post.getPosts().stream()
-                  .allMatch(postEntity -> postEntity.findCategory().endsWith(categoryName));
-            }
-          });
+          .until((Predicate<Post>) post -> post.getPosts().stream()
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostsListPage.java]: MAJOR Rename "post" which hides the field declared at line 34. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java]: MAJOR Rename "driver" which hides the field declared at line 45. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/PostEntity.java [hunk 3]
```diff
@@ -56,7 +53,9 @@ public TopNote findTopNote() {
   }
 
   public String findId() {
-    final String idAttribute = post.findElement(By.className("discussion-more-options")).getAttribute("id");
+    final String
+        idAttribute =
+        post.findElement(By.className("discussion-more-options")).getAttribute("id");
```
Comment (tortila) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/PostEntity.java]: why the line breaks?

File: src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java [hunk 4]
```diff
@@ -486,99 +469,106 @@ public void moderatorOnMobileCanDeleteReportedPostOnPostDetailsPage() {
...
-    PostEntity.Data data = createAndReportAndDeletePostRemotely();
+    final PostEntity.Data data = createAndReportAndDeletePostRemotely();
+    final PostsListPage page = openPostListPageAndWaitUntilLoaded();
+    final PostEntity postEntity = page.getPost().findPostById(data.getId());
+    assertNull(postEntity, NOT_VISIBLE_DELETED_POST_MESSAGE);
+  }
 
-    PostEntity post = openPostsListPage().getPost().findPostById(data.getId());
-    Assertion.assertNull(post, NOT_VISIBLE_DELETED_POST_MESSAGE);
+  private PostsListPage openPostListPageAndWaitUntilLoaded() {
```
Comment (tortila) [inline src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java]: can you please move `openPostListPageAndWaitUntilLoaded` and `openPostDetailsPageAndWaitUntilLoaded` definitions to be next to other helper methods, below all test methods? It took me some time to find them in this 800+ lines file

**Pull request merged**